### DISCUSSION TITLE
On Demand ISR Revalidation Using Edge Functions

### DISCUSSION THREAD

**Discussion by mostafamoussa** [vercel/next.js#51591] [Ideas]
### Goals

1. When revalidating pages in a cron job it's easy to rack up execution time on Vercel, running on Edge would reduce costs

### Background

Running on serverless functions

### Proposal

Having an API that can run on the edge and trigger page revalidation. Yes my team and I are willing to contribute

**Comment by nabildroid** [comment]
Does On demand revalidation works for you in AppRouter?

**Reply by mostafamoussa** [reply]
> Does On demand revalidation works for you in AppRouter? 

We haven't migrated to AppRouter yet. Still using PagesRouter
